tests/unit/test_sharepoint_utils.py

```diff
@@ -0,0 +1,490 @@
+import base64
+import pytest
+import requests
+from unittest.mock import patch, Mock
+from corvus_python.sharepoint.sharepoint_utils import SharePointUtilities
+
+# region Tests for retrieve_image_as_base64
+
+
+def test_retrieve_image_as_base64_success():
+    sharepoint_url = "https://example.sharepoint.com/sites/site_name/library_name/file.jpg"
+    token = "fake_token"
+    mock_drive_id = "fake_drive_id"
+    mock_download_url = "https://example.com/download/file.jpg"
+    mock_image_data = b"fake_image_data"
+    expected_base64_image = base64.b64encode(mock_image_data).decode("utf-8")
+
+    with patch(
+        "corvus_python.sharepoint.sharepoint_utils.SharePointUtilities._get_drive_id", return_value=mock_drive_id
+    ), patch("requests.get") as mock_get:
+
+        # Mock the response for getting the file metadata
+        mock_get.side_effect = [
+            Mock(status_code=200, json=lambda: {"@microsoft.graph.downloadUrl": mock_download_url}),
+            Mock(status_code=200, content=mock_image_data)
+        ]
+
+        result = SharePointUtilities.retrieve_image_as_base64(sharepoint_url, token)
+        SharePointUtilities._get_drive_id.assert_called_once_with(
+            "example.sharepoint.com", "site_name", "library_name", {
+                "Accept": "application/json",
+                "Content-Type": "application/json; charset=utf-8",
+                "Authorization": "Bearer fake_token"
```

Comment:
You have a few instances where you have `"Bearer fake_token"` where most of them use `f"Bearer {token}"`. Might be nice to make this consistent in all tests